keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryUnhappyVetoRepository.kt

```diff
@@ -28,42 +28,42 @@ class InMemoryUnhappyVetoRepository(
 
   private val resources: MutableMap<String, Record> = mutableMapOf()
 
-  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration) {
+  override fun markUnhappyForWaitingTime(resourceId: String, application: String, wait: Duration?) {
     resources[resourceId] = Record(application, calculateExpirationTime(wait))
   }
 
   override fun delete(resourceId: String) {
     resources.remove(resourceId)
   }
 
-  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration): UnhappyVetoStatus {
+  override fun getOrCreateVetoStatus(resourceId: String, application: String, wait: Duration?): UnhappyVetoStatus {
     val record = resources[resourceId]
     if (record == null) {
       resources[resourceId] = Record(application, calculateExpirationTime(wait))
       return UnhappyVetoStatus(shouldSkip = true, shouldRecheck = false)
     }
     return UnhappyVetoStatus(
-      shouldSkip = record.recheckTime > clock.instant(),
-      shouldRecheck = record.recheckTime < clock.instant()
+      shouldSkip = record.recheckTime == null || record.recheckTime > clock.instant(),
+      shouldRecheck = record.recheckTime != null && record.recheckTime < clock.instant()
```

Comment:
I think it would be easier to read if the logic (comparing the recheckTime to now) was encapsulated in a helper function. The same logic is used four times in this file, and I have trouble interpreting what `recheckTime < clock.instant()` means at a glance. Having a semantically meaningful function name instead would help.

Reply:
I have many questions about this implementation, for example: why are there two separate booleans when it seems to me that one would suffice? I can certainly tidy this up but it also feels to me that a slightly more extensive refactor might not be the worst idea.